### ISSUE TITLE
Unable to get raw request body via Body route parameter

### ISSUE THREAD

**Issue by oryba** [tiangolo/fastapi#2460]
### Example

```Python
import uvicorn
from fastapi import Body, FastAPI

app = FastAPI()


@app.post("/test}")
async def update_item(
        payload: str = Body(...)
):
    return {}


uvicorn.run(app, host='127.0.0.1', port=8000)
```

```bash
curl -X POST "http://127.0.0.1:8000/test}" -H  "accept: application/json" -H  "Content-Type: application/text" -d "string"
```

Response: 422 - Unprocessable entity

### Description

Hi there!

I'm trying to pass a text body to my FastAPI endpoint. For sure I can do that [by accessing the request directly](https://fastapi.tiangolo.com/advanced/using-request-directly/). However, from my perspective, the most intuitive way to do that is to use the `Body` like in the example below.

The problem is that Body parameter is always parsed (because of this fragment - [routing.py#L171](https://github.com/tiangolo/fastapi/blob/9e6b069c099d0b63df6e4c754497e0dffbffc5b5/fastapi/routing.py#L171)). How about changing that to something like:

```Python
            if body_field:
                if is_body_form:
                    body = await request.form()
                else:
                    body = await request.body()
                    if isinstance(body_field.type_, (dict, BaseModel)):
                        body = await request.json()
```

Currently `payload: str = Body(...)` always returns validation error, because JSON parsing would never produce a string, so this approach should be fully backward compatible. **I'd gladly create a PR for that if you think this option makes sense:)**

### Environment

* OS: macOS:
* FastAPI Version: 0.61.2
* Python version: 3.8.2

**Comment by ycd** [comment]
What about accessing the raw `request body` from `Request` object?

```python
from fastapi import Request, FastAPI

@app.post("/dummypath")
async def get_body(request: Request):
    return await request.body()
```

**Comment by oryba** [comment]
@ycd yep, as I mentioned above, that's the way to get the raw body. I'm just thinking that it would be great to generalize Body to cover this as well - after all, it's a handy option that for some reason is limited to JSON now

**Comment by hvsummer** [comment]
I saw that if we are using "await request.json()" it might be timeout if we transfer this object into backgroundtask.
how to get the json/raw body without await for it ? 
when the request come to the FastAPI, we should have it at running time, or at least have option to get it without await.
